src/helpers/browsers-config.helper.js

```diff
@@ -82,20 +86,38 @@ const getExtendedBrowsersConfigs = (config) => {
 const browsersConfiguration = (config, commandArgs) => {
   return async () => {
     const browsersSettings = [];
-    const configs = getExtendedBrowsersConfigs(config);
+    const browserConfigs = getExtendedBrowsersConfigs(config, commandArgs);
+    const allSpecs = glob.sync(config.features.map(file => path.join(config.projectPath, file, '**/*.feature'))[0]);
+    const numberOfInstances = (commandArgs.parallel !== undefined && commandArgs.parallel >= allSpecs.length)
+      ? allSpecs.length
+      : (commandArgs.parallel !== undefined && Number.isInteger(commandArgs.parallel) === true && commandArgs.parallel >=! 0)
+        ? commandArgs.parallel
+        : 1;
+    const expectedArrayLength = (numberOfInstances !== 1) ? Math.ceil(allSpecs.length / commandArgs.parallel) : allSpecs.length;
```

Comment:
This one should depend on numberOfInstances value. What's more it does not need to check numberOfInstances because it is at least 1. You cannot get 0 there -> thats why you had previous ifs.

`const expectedArrayLength = Math.ceil(allSpecs.length / numberOfInstances);` should be enough.